keps/20200901-standard-crd.md

```diff
@@ -0,0 +1,272 @@
+```yaml
+---
+title: Standardize ExternalSecret CRD
+authors: all of us
+creation-date: 2020-09-01
+status: draft
+---
+```
+
+# Standardize ExternalSecret CRD
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Terminology](#terminology)
+- [Use-Cases](#use-cases)
+- [Proposal](#proposal)
+  - [API](#api)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This is a proposal to standardize the `ExternalSecret` CRD in an combined effort through all projects that deal with syncing external secrets. This proposal aims to do find the _common denominator_ for all users of `ExternalSecrets`.
+
+## Motivation
+
+There are a lot of different projects in the wild that essentially do the same thing: sync secrets with Kubernetes. The idea is to unify efforts into a single project that serves the needs of all users in this problem space.
+
+As a starting point i would like to define a **common denominator** for a CustomResourceDefinition that serves all known use-cases. This CRD should follow the standard alpha -> beta -> GA feature process.
+
+Once the CRD API is defined we can move on with more delicate discussions about technology, organization and processes.
+
+List of Projects known so far or related:
+* https://github.com/godaddy/kubernetes-external-secrets
+* https://github.com/itscontained/secret-manager
+* https://github.com/ContainerSolutions/externalsecret-operator
+* https://github.com/mumoshu/aws-secret-operator
+* https://github.com/cmattoon/aws-ssm
+* https://github.com/tuenti/secrets-manager
+* https://github.com/kubernetes-sigs/k8s-gsm-tools
+
+### Goals
+
+- Define a alpha CRD
+- Fully document the Spec and use-cases
+
+### Non-Goals
+
+This KEP proposes the CRD Spec and documents the use-cases, not the choice of technology or migration path towards implementing the CRD.
+
+We do not want to sync secrets into a `ConfigMap`.
+
+## Terminology
+
+* Kubernetes External Secrets `KES`: A Application that runs a control loop which syncs secrets
+* KES `instance`: A single entity that runs a control loop.
+* ExternalSecret `ES`: A CustomResource that declares which secrets should be synced
+* Store: Is a **source** for secrets. The Store is external to KES. It can be a hosted service like Alibabacloud SecretsManager, AWS SystemsManager, Azure KeyVault...
```

Comment:
I suggest changing the word `Store` to `Provider` when referring to external "Providers", like AWS SM.

I believe this would avoid confusion with the `SecretStore` CRD.

Any thoughts?

Reply:
An user would say:
- "We need to setup a new `SecretStore` with the `AWSSM` provider", instead of
- "We need to setup a new `SecretStore` with the `AWSSM` store"

Reply:
totally makes sense!

Reply:
fixed in new repo